eng/ApiCompat/ApiCompat.targets

```diff
@@ -0,0 +1,131 @@
+           Condition="!Exists('%(_ContractDependencyDirectories.FullPath)')"/>
+    <Error Text="One ofthe required implementation directories is invalid."
+           Condition="!Exists('%(_DependencyDirectories.FullPath)')"/>
+    <Error Text="Some of Windows Forms assemblies referenced in 'FrameworkListFileClass' group can't be found."
+           Condition="!Exists('%(_WindowsFormsFilePaths.FullPath)')"/>
+
+    <PropertyGroup>
+      <_WindowsFormsApiCompatArgs>"@(_WindowsFormsFilePaths, ',')"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --contract-depends "$(OutputPath),@(_ContractDependencyDirectories, ','),"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --impl-dirs "@(_DependencyDirectories, ','),"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs Condition="'$(CreateBaseline)' != 'true'">$(_WindowsFormsApiCompatArgs) --baseline "$(_WindowsFormsApiCompatBaselineIgnoreFile)"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --ignore-design-time-facades</_WindowsFormsApiCompatArgs>
+
+      <_WindowsFormsCreateBaselineArgs Condition="'$(CreateBaseline)' == 'true'"> &gt; "$(_WindowsFormsApiCompatBaselineIgnoreFile)"</_WindowsFormsCreateBaselineArgs>
+
+      <_WindowsFormsApiCompatResponseFile>$(IntermediateOutputPath)apicompat.winforms.rsp</_WindowsFormsApiCompatResponseFile>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <_WpfApiCompatArgs>"@(_WpfFilePaths, ',')"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs>$(_WpfApiCompatArgs) --contract-depends "$(OutputPath),@(_ContractDependencyDirectories, ','),"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs>$(_WpfApiCompatArgs) --impl-dirs "@(_DependencyDirectories, ','),"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs Condition="'$(CreateBaseline)' != 'true'">$(_WindowsFormsApiCompatArgs) --baseline "$(_WpfApiCompatBaselineIgnoreFile)"</_WpfApiCompatArgs>
+      <_WpfCreateBaselineArgs Condition="'$(CreateBaseline)' == 'true'"> &gt; "$(_WpfApiCompatBaselineIgnoreFile)"</_WpfCreateBaselineArgs>
+
+      <_WpfApiCompatResponseFile>$(IntermediateOutputPath)apicompat.wpf.rsp</_WpfApiCompatResponseFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(IntermediateOutputPath)" />
+    <WriteLinesToFile File="$(_WindowsFormsApiCompatResponseFile)" Lines="$(_WindowsFormsApiCompatArgs)" Overwrite="true" />
+    <WriteLinesToFile File="$(_WpfApiCompatResponseFile)" Lines="$(_WpfApiCompatArgs)" Overwrite="true" />
+  </Target>
+
+  <!--
+    Validate that the current servicing release retains the same public API surface as the first official release.
+  -->
+  <Target Name="EnsureApiCompat"
+          DependsOnTargets="_GetApiCompatInputsAndOutputs"
+          AfterTargets="Build"
+          Inputs="$(_WindowsFormsApiCompatBaselineIgnoreFile);$(_WpfApiCompatBaselineIgnoreFile)"
```

Comment:
I don't think these inputs are sufficient to trigger a rerun when new files are available.